version: '3.9'

services:
  s3api:
    container_name: s3api_api
    build:
      context: .
      target: dev
    ports:
      - '5005:5005'
    env_file:
      - ./.env
    volumes:
      - ./:/app
    networks:
      - s3_net
    depends_on:
      minio:
        condition: service_started
      postgres:
        condition: service_healthy

  minio:
    container_name: s3api_minio
    image: minio/minio:RELEASE.2023-08-23T10-07-06Z.fips
    environment:
      - MINIO_ROOT_USER=user
      - MINIO_ROOT_PASSWORD=password
    command: server /data --console-address ":9001"
    ports:
      - '9000:9000'
      - '9001:9001'
    volumes:
      - ./.data/minio:/data
    networks:
      - s3_net

  postgres:
    container_name: s3api_postgres
    image: postgres:16.1-alpine3.18
    env_file:
      - .env
    ports:
      - '5432:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U $$POSTGRES_USER -d $$POSTGRES_DB']
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - ./.data/postgres:/var/lib/postgresql/data
    networks:
      - s3_net

networks:
  s3_net:
    driver: bridge
